syz-manager: panic in paseExecResult

```
panic: runtime error: slice bounds out of range [4:0]

goroutine 328 [running]:
main.parseExecResult({0x7fd7dd292000?, 0xc02413f410?, 0x7ccc720?})
        /syzkaller/gopath/src/github.com/google/syzkaller/syz-manager/snapshot.go:164 +0xf1
main.(*Manager).snapshotRun(0xc00e2321e0?, 0xc01a64edc0?, 0xc00984fcc0, 0xc01ae80780)
        /syzkaller/gopath/src/github.com/google/syzkaller/syz-manager/snapshot.go:130 +0x2c7
main.(*Manager).snapshotLoop(0xc001b02ea0, {0x20a87e0, 0xc0050a6b90}, 0xc00521c100)
        /syzkaller/gopath/src/github.com/google/syzkaller/syz-manager/snapshot.go:64 +0x28f
main.(*Manager).snapshotInstance(0xc004421df0?, {0x20a87e0, 0xc0050a6b90}, 0x18150e0?, 0xc006112270)
        /syzkaller/gopath/src/github.com/google/syzkaller/syz-manager/snapshot.go:29 +0xb9
github.com/google/syzkaller/vm/dispatcher.(*Pool[...]).runInstance(0x20bfe80, {0x20a87e0, 0xc01e12abe0?}, 0xc00edb1380?)
        /syzkaller/gopath/src/github.com/google/syzkaller/vm/dispatcher/pool.go:126 +0x53b
github.com/google/syzkaller/vm/dispatcher.(*Pool[...]).Loop.func1()
        /syzkaller/gopath/src/github.com/google/syzkaller/vm/dispatcher/pool.go:80 +0x6c
created by github.com/google/syzkaller/vm/dispatcher.(*Pool[...]).Loop in goroutine 1
        /syzkaller/gopath/src/github.com/google/syzkaller/vm/dispatcher/pool.go:78 +0x7e
```

--- a/syz-manager/snapshot.go
+++ b/syz-manager/snapshot.go
@@ -161,6 +161,11 @@ func (mgr *Manager) snapshotRun(inst *vm.Instance, builder *flatbuffers.Builder,
 }
 
 func parseExecResult(data []byte) *flatrpc.ExecResult {
+	if len(data) < flatbuffers.SizeUint32 {
+		return &flatrpc.ExecResult{
+			Error: "the buffer is too small",
+		}
+	}
 	raw, err := flatrpc.Parse[*flatrpc.ExecutorMessageRaw](data[flatbuffers.SizeUint32:])
 	if err != nil {
 		// Don't consider result parsing error as an infrastructure error,

